addons/point_of_sale/models/pos_session.py

```diff
@@ -300,15 +309,534 @@ def _validate_session(self):
         }
 
     def action_pos_session_close(self):
-        # Close CashBox
-        for st in self.statement_ids:
-            if any(abs(st.difference) > self.config_id.amount_authorized_diff for st in self.statement_ids):
-                if not self.user_has_groups("point_of_sale.group_pos_manager"):
-                    raise UserError(_("Your ending balance is too different from the theoretical cash closing (%.2f), the maximum allowed is: %.2f. You can contact your manager to force it.") % (st.difference, self.config_id.amount_authorized_diff))
-                else:
-                    return self._warning_balance_closing()
+        # Session without cash payment method will not have a cash register.
+        # However, there could be other payment methods, thus, session still
+        # needs to be validated.
+        if not self.cash_register_id:
+            return self._validate_session()
+
+        if abs(self.cash_register_difference) > self.config_id.amount_authorized_diff:
+            # Only pos manager can close statements with cash_register_difference greater than amount_authorized_diff.
+            if not self.user_has_groups("point_of_sale.group_pos_manager"):
+                raise UserError(_(
+                    "Your ending balance is too different from the theoretical cash closing (%.2f), "
+                    "the maximum allowed is: %.2f. You can contact your manager to force it."
+                ) % (self.cash_register_difference, self.config_id.amount_authorized_diff))
+            else:
+                return self._warning_balance_closing()
+        else:
+            return self._validate_session()
+
+    def create_account_move(self):
+        """ Create account.move and account.move.line records for this session.
+
+        As a side-effect, the account.move record is set to self.move_id field
+        which is used in the self.reconcile_* functions.
+        """
+        journal = self.config_id.journal_id
+        # Passing default_journal_id for the calculation of default currency of account move
+        # See _get_default_currency in the account/account_move.py.
+        account_move = self.env['account.move'].with_context(default_journal_id=journal.id).create({
+            'journal_id': journal.id,
+            'date': fields.Date.context_today(self),
+            'ref': self.name,
+            'name': journal.sequence_id.next_by_id()
+        })
+        self.write({'move_id': account_move.id})
+        self._create_account_move_lines()
```

Comment:
put content _create_account_move_lines here.
keep the lines of cash
call here 'reconcile_cash_receivable_lines'

Reply:
create the lines with context: 'check_move_validity' = False